app/scripts.babel/contentscript.js

```diff
@@ -50,6 +55,24 @@ window.addEventListener("load", function() {
     var b_desc =      $(this).find(".bz_short_desc_column").first().text().trim();
     var b_changed =   $(this).find(".bz_changeddate_column").first().text().trim();
 
+    // start determining a product's color
+    var b_product_color = "blue"; 
+    if(product_names.indexOf(b_product) == -1) {
+    product_names.push(b_product);
+    while(true) {
+      var random_color = '#'+(Math.random()*0xFFFFFF<<0).toString(16);
```

Comment:
I think having a fixed list of colours, then below at line 66, we get the length of the product colours, then use this as an index to a list of colours. It would also remove the need for this while loop.

This works for now though!